src/os/bluestore/AvlAllocator.cc

```diff
@@ -212,6 +212,11 @@ AvlAllocator::AvlAllocator(CephContext* cct,
   cct(cct)
 {}
 
+AvlAllocator::~AvlAllocator()
+{
+  shutdown();
```

Comment:
@aclamk i don't understand why shall we call `shutdown()` here. because in an ideal world, the allocated memory chunks are supposed to be returned to allocator when the program is shutting down. if there is any tracked range still left in allocator, that's a bug. so shall we instead put
```c++
AvlAllocator::~AvlAllocator()
{
  assert(range_size_tree.empty());
  assert(range_tree.empty());
}
```
? to verify the program is well behaved ?

Reply:
@tchaikov Your proposal is to make invocation of `shutdown()` before destructor mandatory?

I agree with one thing - I was surprised to see that when I destroy AvlAllocator mempool did not go to 0. I added shutdown and achieved expected value of 0.
On that I drove conclusion that AVL's data structures do contain raw pointers that are not automatically freed in default destructor. I did not even investigated what it actually was.

Reply:
> @tchaikov Your proposal is to make invocation of `shutdown()` before destructor mandatory?
> 

i don't think we will need to call `shutdown()` , unless we are sure that we have bug somewhere, and we want to cover it up.

> I agree with one thing - I was surprised to see that when I destroy AvlAllocator mempool did not go to 0. I added shutdown and achieved expected value of 0.
> On that I drove conclusion that AVL's data structures do contain raw pointers that are not automatically freed in default destructor. I did not even investigated what it actually was.

Reply:
@tchaikov  I verified that problem is really related to memory leak.
`valgrind ./bin/unittest_alloc_bench --gtest_filter=\*mempool\*2
....
==25115== LEAK SUMMARY:
==25115==    definitely lost: 497,440 bytes in 6,218 blocks`